### PR TITLE
[Feature] support dayofweek_iso function

### PR DESCRIPTION
**Pull request by xuzifu666**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by xuzifu666** [comment]
cc @silverbullet233

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by silverbullet233** [comment]
need some test cases, you can refer to https://github.com/StarRocks/starrocks/tree/main/test and add them into `test/sql` dir.

**Review by murphyatwork — DISMISSED** [review]

File: be/src/exprs/time_functions.h [hunk 1]
```diff
@@ -103,6 +103,21 @@ class TimeFunctions {
      */
     DEFINE_VECTORIZED_FN(day_of_week);
 
+    /**
```
Comment (chaoyli) [inline be/src/exprs/time_functions.h:106]: Could you give a syntax and example in the commit message?
It's better to describe how to use it in the commit message.

Comment (xuzifu666) [reply be/src/exprs/time_functions.h:106]: ok，i would add these message and test

**Comment by silverbullet233** [comment]
btw，better to add a document for this function too. you can refer to : https://github.com/StarRocks/starrocks/tree/main/docs/sql-reference/sql-functions/date-time-functions

**Comment by xuzifu666** [comment]
> btw，better to add a document for this function too. you can refer to : https://github.com/StarRocks/starrocks/tree/main/docs/sql-reference/sql-functions/date-time-functions

yeah，get it

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docs/sql-reference/sql-functions/date-time-functions/week_iso.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ The date parameter must be of the DATE or DATETIME type.
 ## Syntax
 
 ```Haskell
-INT WEEK_ISO(DATETIME date)
+INT DAY_OF_WEEK(DATETIME date)
```
Comment (silverbullet233) [inline docs/sql-reference/sql-functions/date-time-functions/week_iso.md:12]: why change here?

File: docs/sql-reference/sql-functions/date-time-functions/day_of_week_iso.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# week_iso
```
Comment (silverbullet233) [inline docs/sql-reference/sql-functions/date-time-functions/day_of_week_iso.md:1]: title is mismatched

File: gensrc/script/functions.py [hunk 4]
```diff
@@ -183,8 +183,8 @@
 
     [10320, "conv", "VARCHAR", ["BIGINT", "TINYINT", "TINYINT"], "MathFunctions::conv_int"],
     [10321, "conv", "VARCHAR", ["VARCHAR", "TINYINT", "TINYINT"], "MathFunctions::conv_string"],
-
-    [10322, "square", "DOUBLE", ["DOUBLE"], "MathFunctions::square"],
+    
+    [10322, "square", "DOUBLE", ["DOUBLE"], "MathFunctions::square"],
```
Comment (silverbullet233) [inline gensrc/script/functions.py:188]: format should be fixed

Comment (xuzifu666) [reply gensrc/script/functions.py:188]: ok, would fix it

File: docs/sql-reference/sql-functions/date-time-functions/week_iso.md (hunk 2 above)
Comment (xuzifu666) [reply docs/sql-reference/sql-functions/date-time-functions/week_iso.md:12]: my misstake，would recover it

File: docs/sql-reference/sql-functions/date-time-functions/day_of_week_iso.md (hunk 3 above)
Comment (xuzifu666) [reply docs/sql-reference/sql-functions/date-time-functions/day_of_week_iso.md]: done，thanks

File: docs/sql-reference/sql-functions/date-time-functions/day_of_week_iso.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+```Haskell
+INT DAY_OF_WEEK_ISO(DATETIME date)
+```
+
+## Examples
+
+```Plain Text
+MySQL > select dayofweek_iso('2008-02-20 00:00:00');
++-----------------------------------+
+| week_iso('2008-02-20 00:00:00')   |
```
Comment (silverbullet233) [inline docs/sql-reference/sql-functions/date-time-functions/day_of_week_iso.md:20]: column name in result is mismatched

Comment (xuzifu666) [reply docs/sql-reference/sql-functions/date-time-functions/day_of_week_iso.md]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by silverbullet233** [comment]
move to https://github.com/StarRocks/starrocks/pull/26240, so close it.

**Pull request closed without merging**